napari/layers/image/tests/test_pyramid.py

```diff
@@ -292,8 +292,24 @@ def test_xml_list():
     shapes = [(40, 20), (20, 10), (10, 5)]
     np.random.seed(0)
     data = [np.random.random(s) for s in shapes]
-    layer = Pyramid(data)
+    layer = Image(data, pyramid=True)
     xml = layer.to_xml_list()
     assert type(xml) == list
     assert len(xml) == 1
     assert type(xml[0]) == Element
+
+
+def test_create_random_pyramid():
+    """Test instantiating Image layer with random 2D data."""
+    shape = (20_000, 20)
+    np.random.seed(0)
+    data = np.random.random(shape)
+    layer = Image(data)
+    assert np.all(layer.data == data)
+    assert layer.pyramid == True
+    assert layer._data_pyramid[0].shape == shape
+    assert layer._data_pyramid[1].shape == (shape[0] / 2, shape[1])
+    assert layer.ndim == len(shape)
+    assert layer.shape == shape
+    assert layer.rgb == False
+    assert layer._data_view.ndim == 2
```

Comment:
maybe add a test for `test_infer_pyramid` that you pass in a list of decreasing arrays to, without `pyramid=True`, and it should correctly set `layer.[is_]pyramid`?

Reply:
yup, good catch, done